apps/mailinglists/tests.py

```diff
@@ -0,0 +1,160 @@
+            email_local_part="gloshaugen",
+            domain=MailGroup.Domains.ONLINE_NTNU_NO,
+            description="En haug i Trondheim",
+        )
+
+        mail = MailEntity.objects.get(name=mailinglist.name)
+
+        self.assertEqual(mail.description, mailinglist.description)
+        self.assertEqual(mail.email, mailinglist.email)
+        self.assertEqual(mail.public, mailinglist.public)
+
+    def test_anon_api_call(self):
+        # FIXME: Make this test pass by filtering out maillist's non-public emails
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data["results"],
+            [
+                OrderedDict(
+                    [
+                        ("id", 1),
+                        ("email", "gresk@linjeforeninger.no"),
+                        ("name", "Greske linjeforeninger"),
+                        ("description", "Linjeforenigner med greske navn."),
+                        ("public", True),
+                        (
+                            "members",
+                            [
+                                OrderedDict(
+                                    [
+                                        ("id", 1),
+                                        ("email", "hs@beta.ntnu.no"),
+                                        ("name", "Beta Linjeforening"),
+                                        (
+                                            "description",
+                                            "Vi er kanskje ikke Alfa, men vi er bedre enn Gamma.",
+                                        ),
+                                        ("public", True),
+                                    ]
+                                )
+                            ],
+                        ),
+                    ]
+                )
+            ],
+        )
+
+    def test_authenticated_api_call(self):
+        # FIXME: Make this test pass
+        content_type = ContentType.objects.get_for_model(MailGroup)
+        permission = Permission.objects.get(
+            codename="view_mailgroup", content_type=content_type
+        )
+        self.user.user_permissions.add(permission)
+
+        self.assertTrue(self.user.has_perm("mailinglists.view_mailgroup"))
+
+        response = self.client.get(self.url, {**self.generate_headers()})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.data["results"],
+            [
+                OrderedDict(
+                    [
+                        ("id", 1),
+                        ("email", "gresk@linjeforeninger.no"),
+                        ("name", "Greske linjeforeninger"),
+                        ("description", "Linjeforenigner med greske navn."),
+                        ("public", True),
+                        (
+                            "members",
```

Comment:
What do you suggest we do instead? I am rather new to API-testing, so would appreciate feedback on good practices 😄

Reply:
I think it is OK to just check the status code, and maybe check a specific field in some very specific cases where serialization is very complex or dependent on how you do the request. Like if a field is a specific thing when you are logged in and other of you are not.

Otherwise serializer behavior us very predictable, and the only thing you end up doing micro-managing the test data when you are doing changes :stuck_out_tongue: